File: cracking_the_coding_interview/algorithms/sorting_bubble_sort.py
n = int(input().strip())
a = list(map(int, input().strip().split(' ')))

swaps = 0
for i in range(n):
    for j in range(n - 1):
        if(a[j] > a[j + 1]):
            a[j], a[j + 1] = a[j + 1], a[j]
            swaps += 1

print('Array is sorted in {} swaps.'.format(swaps))
print('First Element: {}'.format(a.pop(0)))
print('Last Element: {}'.format(a.pop()))

File: cracking_the_coding_interview/algorithms/sorting_bubble_sort.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {
  static void countSwaps(int[] a) {
      int swaps = 0;
      for(int i = 0; i < a.Length; i++) {
          for(int j = i + 1; j < a.Length; j++){
              if(a[i] > a[j]){
                  int temp = a[i];
                  a[i] = a[j];
                  a[j] = temp;
                  swaps++;
              }
          }
      }
      
      Console.WriteLine("Array is sorted in " + swaps + " swaps.");
      Console.WriteLine("First Element: " + a[0]);
      Console.WriteLine("Last Element: " + a[a.Length - 1]);
  }
}